app.js

```diff
@@ -32,3 +33,268 @@ setMobileClass();
+  cardLanguage.appendChild(btnSmall3);
+
+  //  creating card-cta element
+  const cardCta = document.createElement('a');
+  cardCta.classList.add('card-cta');
+  cardCta.innerText = cards[i].cta;
+  cardContent.appendChild(cardCta);
+
+  cardCta.addEventListener('click', () => {
+    const modal = document.createElement('div');
+    modal.classList.add('modal');
+    modal.style.display = 'block';
+    body.appendChild(modal);
+
+    // creating modal-content element
+    const modelContent = document.createElement('div');
+    modelContent.classList.add('modal-content');
+    modal.appendChild(modelContent);
+
+    // crating close element
+    const close = document.createElement('span');
+    close.classList.add('close');
+    close.innerHTML = '&times;';
+    close.addEventListener('click', () => {
+      body.removeChild(modal);
+    });
+    modelContent.appendChild(close);
+
+    // creating modalTitle element
+    const modalTitle = document.createElement('h2');
+    modalTitle.classList.add('modal-title');
+    modalTitle.innerText = cards[i].title;
+    modelContent.appendChild(modalTitle);
+
+    // ok
+    modelContent.appendChild(cardDescription);
+
+    // creating a images
+    if (window.innerWidth > MOBILE_WIDTH) {
+      const img = document.createElement('img');
+      img.src = cards[i].imgDesktop;
+      img.alt = cards[i].title;
+      img.style.marginBottom = '2rem';
+      img.classList.add('modal-img');
+      modelContent.appendChild(img);
+    } else {
+      const img = document.createElement('img');
+      img.src = cards[i].imgMabail;
+      img.alt = cards[i].title;
+      img.style.marginBottom = '1rem';
+      img.classList.add('modal-img');
+      modelContent.appendChild(img);
+    }
+
+    // creating model-info element
+    const modelInfo = document.createElement('div');
+    modelInfo.classList.add('model-info');
+    modelContent.appendChild(modelInfo);
+
+    if (window.innerWidth > MOBILE_WIDTH) {
+      modelInfo.style.display = 'flex';
+    }
+
+    // creating model-paragraph element
+    const modelParagraph = document.createElement('p');
+    modelParagraph.innerText = cards[i].paragraph;
+    modelInfo.appendChild(modelParagraph);
+
+    if (window.innerWidth > MOBILE_WIDTH) {
+      modelParagraph.classList.add('model-paragraph');
+    }
+
+    // creating model language
+    const modelinfoActive = document.createElement('div');
+    modelInfo.appendChild(modelinfoActive);
+    modelinfoActive.appendChild(cardLanguage);
+
+    // model cta
+    const cardCta1 = document.createElement('a');
+    cardCta1.classList.add('card-cta');
+    cardCta1.classList.add('model-cta');
+    cardCta1.innerText = 'See live';
+    modelinfoActive.appendChild(cardCta1);
+
+    const cardCta2 = document.createElement('a');
+    cardCta2.classList.add('card-cta');
+    cardCta2.innerText = 'See source';
+    modelinfoActive.appendChild(cardCta2);
+  });
+
+  grid.appendChild(card);
+}
```

Comment:
- [ OPTIONAL]  kindly avoid very large functions. You can always break functions into small manageable parts. Large functions are hard to maintain and debug. What do you think 🤔 ?